packages/edit-site/src/components/header/index.js

```diff
@@ -70,10 +70,9 @@ export default function Header( { openEntitiesSavedStates } ) {
 	const displayBlockToolbar =
 		! isLargeViewport || deviceType !== 'Desktop' || hasFixedToolbar;
 
-	let { title } = getTemplateInfo( template );
-	if ( 'wp_template_part' === templateType ) {
-		title = templatePart?.slug;
-	}
+	const { title } = useTemplateInfo(
+		templateType === 'wp_template' ? template : templatePart
```

Comment:
I personally think it's odd to let `useTemplateInfo` accept a template part. Is that just to support this line of code? I feel like it makes the hook do more than one thing

Reply:
True dat!
It always sounded awkward to me too, but didn't think of splitting the hooks for templates and parts.
Will do it!

Reply:
Unless you solve the issue at the higher level and use a different component (`TemplateNavigationItem` vs `TemplatePartNavigationItem`) depending on if there is a template or a template part provided.

I personally second what @noahtallen shared. As of today, those two concepts have very similar characteristics but for some reason (that I want to learn about in the future), they are two independent CPTs on the server.  Based solely on that, I believe there is a reason that they should also be treated as such on the fronted.

Reply:
Ah, it has been so many years since we discussed having one or more CPTs, and my memory is not exactly great. 😅 

While I agree that these two entities should be treated separately in the front end, I'd also argue that `useTemplateInfo` is for the most part a conversion of `getTemplateInfo` which was also designed to handle both templates and template parts.
I'd say a different PR or a follow up to this would be a better place to clear up that confusion.

Reply:
> getTemplateInfo which was also designed to handle both templates and template parts.

Oh, I thought we only used it for templates! At least, in the FSE header area, we only used it for templates and did template parts with something else

Reply:
You know I think you are right!
I'll see to move the template part handling out of the way.